gateway/gateway.go

```diff
@@ -1366,51 +1389,56 @@ func (gateway *HandleT) StartWebHandler(ctx context.Context) error {
 	gateway.logger.Infof("WebHandler Starting on %d", webPort)
 
 	srvMux := mux.NewRouter()
-	srvMux.Use(
-		middleware.StatMiddleware(ctx),
-		middleware.LimitConcurrentRequests(maxConcurrentRequests),
-		middleware.ContentType(),
-	)
-	srvMux.HandleFunc("/v1/batch", gateway.webBatchHandler).Methods("POST")
-	srvMux.HandleFunc("/v1/identify", gateway.webIdentifyHandler).Methods("POST")
-	srvMux.HandleFunc("/v1/track", gateway.webTrackHandler).Methods("POST")
-	srvMux.HandleFunc("/v1/page", gateway.webPageHandler).Methods("POST")
-	srvMux.HandleFunc("/v1/screen", gateway.webScreenHandler).Methods("POST")
-	srvMux.HandleFunc("/v1/alias", gateway.webAliasHandler).Methods("POST")
-	srvMux.HandleFunc("/v1/merge", gateway.webMergeHandler).Methods("POST")
-	srvMux.HandleFunc("/v1/group", gateway.webGroupHandler).Methods("POST")
-	srvMux.HandleFunc("/health", app.LivenessHandler(gateway.jobsDB)).Methods("GET")
-	srvMux.HandleFunc("/", app.LivenessHandler(gateway.jobsDB)).Methods("GET")
-	srvMux.HandleFunc("/v1/import", gateway.webImportHandler).Methods("POST")
-	srvMux.HandleFunc("/v1/audiencelist", gateway.webAudienceListHandler).Methods("POST")
-	srvMux.HandleFunc("/pixel/v1/track", gateway.pixelTrackHandler).Methods("GET")
-	srvMux.HandleFunc("/pixel/v1/page", gateway.pixelPageHandler).Methods("GET")
-	srvMux.HandleFunc("/v1/webhook", gateway.webhookHandler.RequestHandler).Methods("POST", "GET")
-	srvMux.HandleFunc("/beacon/v1/batch", gateway.beaconBatchHandler).Methods("POST")
-
-	srvMux.HandleFunc("/version", gateway.versionHandler).Methods("GET")
-	srvMux.HandleFunc("/robots.txt", gateway.robots).Methods("GET")
+	jsonMux, err := newSubRouter(ctx, srvMux, response.ContentTypeJSON)
+	if err != nil {
+		return err
+	}
+	plainMux, err := newSubRouter(ctx, srvMux, response.ContentTypePlain)
+	if err != nil {
+		return err
+	}
+
+	plainMux.HandleFunc("/v1/batch", gateway.webBatchHandler).Methods("POST")
+	plainMux.HandleFunc("/v1/identify", gateway.webIdentifyHandler).Methods("POST")
+	plainMux.HandleFunc("/v1/track", gateway.webTrackHandler).Methods("POST")
+	plainMux.HandleFunc("/v1/page", gateway.webPageHandler).Methods("POST")
+	plainMux.HandleFunc("/v1/screen", gateway.webScreenHandler).Methods("POST")
+	plainMux.HandleFunc("/v1/alias", gateway.webAliasHandler).Methods("POST")
+	plainMux.HandleFunc("/v1/merge", gateway.webMergeHandler).Methods("POST")
+	plainMux.HandleFunc("/v1/group", gateway.webGroupHandler).Methods("POST")
+	plainMux.HandleFunc("/v1/import", gateway.webImportHandler).Methods("POST")
+	plainMux.HandleFunc("/v1/audiencelist", gateway.webAudienceListHandler).Methods("POST")
+	plainMux.HandleFunc("/v1/webhook", gateway.webhookHandler.RequestHandler).Methods("POST", "GET")
+	plainMux.HandleFunc("/beacon/v1/batch", gateway.beaconBatchHandler).Methods("POST")
+	plainMux.HandleFunc("/pixel/v1/track", gateway.pixelTrackHandler).Methods("GET")
```

Comment:
Create another content type (image/gif) for /pixel/* endpoints.

Right now, content type for these endpoints is set in the gateway code.
https://github.com/rudderlabs/rudder-server/blob/bced8d58bf8d2a880c41f9f26e965d64c1efdd76/gateway/gateway.go#L1295